frontend/src/lib/gql/typeahead-queries.ts

```diff
@@ -0,0 +1,51 @@
+import type { LoadAdminUsersTypeaheadQuery, UserFilterInput } from './types';
+
+import { getClient } from './gql-client';
+import { graphql } from './generated';
+import { DEFAULT_PAGE_SIZE } from '$lib/components/Paging';
+
+export function userFilter(userSearch: string): UserFilterInput {
+  return {
+    or: [
+      {name: {icontains: userSearch}},
+      {email: {icontains: userSearch}},
+      {username: {icontains: userSearch}}
+    ]
+  };
+}
+
+export type UserTypeaheadResult = NonNullable<NonNullable<LoadAdminUsersTypeaheadQuery['users']>['items']>;
+export type SingleUserTypeaheadResult = UserTypeaheadResult[number];
+
+export async function _typeaheadSearch(userSearch: string, limit = DEFAULT_PAGE_SIZE): Promise<UserTypeaheadResult> {
+  if (!userSearch) return Promise.resolve([]);
+  const client = getClient();
+  const result = client.query(graphql(`
+    query loadAdminUsersTypeahead($filter: UserFilterInput, $take: Int!) {
+      users(
+        where: $filter, orderBy: {name: ASC}, take: $take) {
+        totalCount
+        items {
+          id
+          name
+          email
+          username
+        }
+      }
+    }
+  `), { filter: userFilter(userSearch), take: limit+1 });
```

Comment:
My idea is that if the filter is returning more results than the desired number, we don't display any results. I.e., as you type `robin_munn`, when you type "r" there will be hundreds of results, and it's very unlikely that what you want is in the first 10. (Assuming that I set the default limit to 10, which I'm going to do.) So if there are 11 or more results, then we're going to not show any.

So I wrote the GQL query to return a max `limit+1` results, in this case it'll be 11. The logic later on only shows results if there are `<= limit` of them, so if 11 results were returned, we don't show the results because there were probably hundreds. But if there are 10 or less, then we know that the one the admin actually wants is in the list, so we show the list.

Reply:
> My idea is that if the filter is returning more results than the desired number, we don't display any results.

How can the user distinguish between "no results" and "too many results"?

In my opinion, I think showing the top ten matching users sorted by "user created date" would be better.  It could be more relevant than default database sort.

Reply:
I don't quite get it. It seems to me that your logic will work just fine without `+1`. You can (and I think you actually are) determine if there are more results using the returned `totalCount`.
I.e. `if (totalCount > limit)` you want to not show anything. Isn't that what you're going for?

Reply:
You know, I just assumed that `totalCount` was "how many items were returned by this query" rather than "this is how many items you *would* have gotten if you hadn't specified a `limit`". Of course it's actually the latter, in which case the +1 logic I used doesn't make sense. (And I didn't have a quick way of verifying that since there are only four users in the default test data, so totalCount was always less than 10 unless I went to the bulk-add tool and added a bunch more users).

Yeah, I'll remove the +1 since there are much simpler ways to do this. Now that I've stopped having a big oops moment about the meaning of `totalCount`... :-)

Reply:
Done in commit 948ae868.